infrastructure/templates/public-api/main.bicep

```diff
@@ -252,9 +249,19 @@ module apiContainerAppModule 'components/containerApp.bicep' = if (psqlDbUsersAd
         name: 'DataProcessorFunctionAppIdentityName'
         value: dataProcessorFunctionAppFullName
       }
+      {
+        // This property informs the Container App of the name of the Publisher's system-assigned identity.
+        // It uses this to grant permissions to the Publisher user in order for it to be able to access
+        // tables in the "public_data" database successfully.
+        name: 'PublisherFunctionAppIdentityName'
+        value: publisherFunctionAppFullName
+      }
     ]
     tagValues: tagValues
   }
+  dependsOn: [
+    postgreSqlServerModule
+  ]
```

Comment:
Just wondering if this is still required? If deployContainerApp is true and updatePsqlFlexibleServer is false would this work?

Reply:
Yeah, so this is in the situation where `updatePsqlFlexibleServer` is false.  In that case, it's dependent on `postgreSqlServerModule` being complete before continuing - we have to be explicit about the dependency because we're not using any actual values from the `postgreSqlServerModule` output in this definition.

If `updatePsqlFlexibleServer` is true, `postgreSqlServerModule` will be undefined, and any undefined references in the `dependsOn` array get ignored, so this module can just deploy immediately.